### PR TITLE
Add support for graphql-transport-ws single result operations

### LINKED ISSUE #1638: Query and Mutation not accepted over protocol `graphql-protocol-ws`
Attempting to issue a `query` over the websockets protocol results in the socket being closed by the server with error 4400.
However, the `websockets-transport-ws` protocol specifically allows for queries and mutations, **in addition to** subscriptions: https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#query-and-mutation-operations-without-streaming-directives

It would be useful to have these supported, so that a client need only keep a single websocket connection open and not have to have a separate HTTP pathway for queries and mutations.

### PR DESCRIPTION
**Pull request by DoctorJohn**
## Description

In addition to subscriptions the `graphql-transport-ws` [protocol](https://github.com/enisdenjo/graphql-ws/blob/master/PROTOCOL.md#single-result-operation) specifies so called single result operations (i.e. mutations and queries over the websocket connection). Strawberry currently does not support single result operations, this PR adds support for them.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* Closes #1638

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

This release adds support for `graphql-transport-ws` single result operations.

Single result operations allow clients to execute queries and mutations over an existing WebSocket connection.


---
Here's the preview release card for twitter:




Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Jonathan Ehwald for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

**Review by patrick91 — COMMENTED** [review]
Looks great! Should we update the docs for this?

Also is there any example of clients using websockets for single operations?

File: strawberry/utils/get_operation_type.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+from typing import Optional, cast
```
[the remark this answers is not in the document]
Comment (patrick91) [reply strawberry/utils/get_operation_type.py:1]: I don't think we have defined a style guide for things like this and I personally don't really have a strong preference :) maybe @BryceBeagle has some suggestions too :)

**Comment by DoctorJohn** [comment]
> Looks great! Should we update the docs for this?

Technically there is not much to say about it, it just works out of the box and is part of the `graphql-transport-ws` protocol which we mention in the docs. Maybe we could add a paragraph teasing the feature :)

> Also is there any example of clients using websockets for single operations?

The specifications repo contains a runnable [client example](https://github.com/enisdenjo/graphql-ws#use-the-client) and [recipes](https://github.com/enisdenjo/graphql-ws#recipes) how to make it work with frameworks such as Apollo.

File: strawberry/utils/get_operation_type.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (DoctorJohn) [reply strawberry/utils/get_operation_type.py:1]: TBH I didn't really know were to put those helpers after I extracted them from the ExecutionContext to use them elsewhere. However, I somewhat prefer having functions and especially classes in separate files. That way I can easily figure out where to import them from looking at the file tree. But that's just a personal preference, I don't mind putting them somewhere else :)

**Review by patrick91 — APPROVED** [review]

**Comment by kristjanvalur** [comment]
Good stuff.  One question, in case I end up contributing, I noticed a lot of code duplication in the tests for different integrations (aiohttp, fastapi, etc), is this necessary, and in this case, can they be factored out somehow?

**Comment by DoctorJohn** [comment]
@kristjanvalur Technically it would be enough to test the ASGI and AIOHTTP integrations, since fastapi and django use the ASGI integration. Combining both ASGI and AIOHTTP tests could be tricky, mostly because they have different test clients.

**Pull request merged**